### ISSUE TITLE
Convert parallel coordinates into FeatureVisualizer

### ISSUE THREAD

**Issue by rebeccabilbro** [DistrictDataLabs/yellowbrick#50]
Convert parallel coordinates into a `FeatureVisualizer`. Allow the user to specify which dimensions to visualize, either in `init` or `draw`.

**Comment by bbengfort** [comment]
I'll take this right now and instead of using pandas, I'll grab and improve upon the implementation given here: 

https://github.com/pydata/pandas/blob/master/pandas/tools/plotting.py

**Comment by bbengfort** [comment]
A couple of notes:

1. How do we most efficiently get the unique feature names from a numpy array?
2. How do we deal with DataFrame vs. Numpy Array (and not have pandas as a dependency)?
3. How can we limit the number of features?

For no. 3: you can pass in a list of feature names, but this must be the same dimension as the number of columns of X. There needs to be a way to map column index to feature name to filter which features we select.

**Comment by bbengfort** [comment]
This could be useful (color selector):

https://github.com/pydata/pandas/blob/master/pandas/tools/plotting.py#L163

**Comment by bbengfort** [comment]
Ok, so this is the question - how do we play nice with both Pandas data frames and the normal numpy arrays that Scikit-Learn expects?

**Comment by rebeccabilbro** [comment]
Hmm... I found [this](https://github.com/paulgb/sklearn-pandas/blob/master/sklearn_pandas/dataframe_mapper.py), which seemed interesting.

Is the real issue that we don't want Pandas as a dependency?

**Comment by bbengfort** [comment]
@rebeccabilbro I really don't feel comfortable adding any dependencies other than Scikit-Learn and matplotlib. Pandas also has to be built with Cython (though they have their wheels in good shape); but that could lead to trickiness down the road. Sklearn [depends on SciPy and NumPy](https://github.com/scikit-learn/scikit-learn/blob/master/setup.py#L246) but not much else. 

That said, it does seem that Scikit-Learn and Pandas DataFrames do play nice together, most of the time. I have the following answers to the Scikit-Learn/Pandas Integration:

1. They don't go together, use `df.values()` or `df.as_matrix()` https://www.quora.com/How-does-python-pandas-go-along-with-scikit-learn-library

2. They go great because a DataFrame is basically a matrix http://machinelearningmastery.com/prepare-data-machine-learning-python-scikit-learn/

I did see the @paulgb project on the topic - but it appears that Scikit-Learn >= 0.17 plays nice with DataFrames - but I can't find a reason for that (on the Scikit-Learn side, maybe it's on the Pandas side?); that said a [search for DataFrame](https://github.com/scikit-learn/scikit-learn/search?p=1&q=DataFrame&utf8=%E2%9C%93) in the sklearn source reveals that it is tested against.

So right now, I'm going to create an is_dataframe function that will internally import Pandas to do type checking, otherwise it will go about it in a less fun way. I will handle the DataFrame doing conversions as necessary, then behave like a regular transformer.

**Comment by bbengfort** [comment]
Also this: 
http://scikit-learn.org/stable/auto_examples/hetero_feature_union.html

**Comment by bbengfort** [comment]
So I'm going to go ahead and close this since we have a visualizer that spits out parallel coordinates; but I'll put a new issue in to get the DataFrame integration as well as other items like feature limits and normalization.

**Comment by bbengfort** [comment]
Adding an example for anyone who is interested!

![yb_parallel_coords_palette](https://cloud.githubusercontent.com/assets/745966/19199697/46b41f24-8c93-11e6-98a9-417d3698c07c.png)
